src/NServiceBus.RabbitMQ/RabbitMQTransportInfrastructure.cs

```diff
@@ -25,22 +25,17 @@ public RabbitMQTransportInfrastructure(SettingsHolder settings, string connectio
             var connectionConfiguration = new ConnectionStringParser(settings).Parse(connectionString);
             connectionFactory = new ConnectionFactory(connectionConfiguration);
 
-            SetUsePublisherConfirmsIfPresent(settings, connectionConfiguration);
-
             CreateTopology();
 
-            channelProvider = new ChannelProvider(connectionFactory, routingTopology, connectionConfiguration.UsePublisherConfirms);
-
-            RequireOutboxConsent = false;
-        }
-
-        static void SetUsePublisherConfirmsIfPresent(SettingsHolder settings, ConnectionConfiguration connectionConfiguration)
-        {
-            if (settings.HasSetting(SettingsKeys.UsePublisherConfirms))
+            bool usePublisherConfirms;
+            if (!settings.TryGet(SettingsKeys.UsePublisherConfirms, out usePublisherConfirms))
             {
-                var publisherCofirms = settings.Get<bool>(SettingsKeys.UsePublisherConfirms);
-                connectionConfiguration.UsePublisherConfirms = publisherCofirms;
+                usePublisherConfirms = true;
             }
+
+            channelProvider = new ChannelProvider(connectionFactory, routingTopology, usePublisherConfirms);
+
+            RequireOutboxConsent = false;
         }
```

Comment:
At this point it seems rather convoluted to me to have this as a separate private method, but I'm happy to go with it for now.